vm-setup/roles/v1aX_integration_test/tasks/main.yml

```diff
@@ -40,9 +40,34 @@
     namespace: "{{ NAMESPACE }}"
   when: v1aX_integration_test_action in provision_workers_actions
 
-- name: verify deployment
-  include_tasks: verify.yml
+- name: Verify provisioning
+  block:
+    - name: verify deployment
+      include_tasks: verify.yml
   when: v1aX_integration_test_action in verify_actions
+  rescue:
+    - name: Gather debug info
+      community.kubernetes.k8s_info:
+        api_version: "{{ item.api_version }}"
+        kind: "{{ item.kind }}"
+      register: resources
+      loop:
+        - api_version: cluster.x-k8s.io/v1alpha3
+          kind: Cluster
+        - api_version: metal3.io/v1alpha1
+          kind: BareMetalHost
+        - api_version: cluster.x-k8s.io/v1alpha3
+          kind: Machine
+        - api_version: v1
+          kind: Node
+        - api_version: v1
+          kind: Pod
+    - name: Print debug info
+      debug:
+        msg: "{{ resources }}"
+    - name: Fail due to earlier error
+      ansible.builtin.fail:
+        msg: Failed due to earlier errors
```

Comment:
This is an example of what we could do to gather more information in case some task fails. The `rescue` part is run if there is a failure in the `block`.